presto-hive/src/main/java/com/facebook/presto/hive/util/Statistics.java

```diff
@@ -134,6 +323,139 @@ private static long convertLocalToUtc(DateTimeZone timeZone, long value)
+        if (computedStatistics.containsKey(NUMBER_OF_DISTINCT_VALUES)) {
+            result.setDistinctValuesCount(BIGINT.getLong(computedStatistics.get(NUMBER_OF_DISTINCT_VALUES), 0));
+        }
+
+        // DATA SIZE
+        if (computedStatistics.containsKey(MAX_VALUE_SIZE_IN_BYTES)) {
+            result.setMaxColumnLength(BIGINT.getLong(computedStatistics.get(MAX_VALUE_SIZE_IN_BYTES), 0));
+        }
+        if (computedStatistics.containsKey(AVERAGE_VALUE_SIZE_IN_BYTES)) {
+            result.setAverageColumnLength(DOUBLE.getDouble(computedStatistics.get(AVERAGE_VALUE_SIZE_IN_BYTES), 0));
+        }
+
+        // NUMBER OF NULLS
+        if (computedStatistics.containsKey(NUMBER_OF_NON_NULL_VALUES)) {
+            result.setNullsCount(rowCount - BIGINT.getLong(computedStatistics.get(NUMBER_OF_NON_NULL_VALUES), 0));
+        }
+
+        // NUMBER OF FALSE, NUMBER OF TRUE
+        if (computedStatistics.containsKey(NUMBER_OF_TRUE_VALUES) && computedStatistics.containsKey(NUMBER_OF_NON_NULL_VALUES)) {
+            long numberOfTrue = BIGINT.getLong(computedStatistics.get(NUMBER_OF_TRUE_VALUES), 0);
+            long numberOfNonNullValues = BIGINT.getLong(computedStatistics.get(NUMBER_OF_NON_NULL_VALUES), 0);
+            result.setBooleanStatistics(new BooleanStatistics(OptionalLong.of(numberOfTrue), OptionalLong.of(numberOfNonNullValues - numberOfTrue)));
+        }
+        return result.build();
+    }
+
+    private static void setMinMax(ConnectorSession session, DateTimeZone timeZone, Type type, Block min, Block max, HiveColumnStatistics.Builder result)
+    {
+        if (type.equals(BIGINT) || type.equals(INTEGER) || type.equals(SMALLINT) || type.equals(TINYINT)) {
+            result.setIntegerStatistics(new IntegerStatistics(getIntegerValue(session, type, min), getIntegerValue(session, type, max)));
+        }
+        else if (type.equals(DOUBLE) || type.equals(REAL)) {
+            result.setDoubleStatistics(new DoubleStatistics(getDoubleValue(session, type, min), getDoubleValue(session, type, max)));
+        }
+        else if (type.equals(DATE)) {
+            result.setDateStatistics(new DateStatistics(getDateValue(session, type, min), getDateValue(session, type, max)));
+        }
+        else if (type.equals(TIMESTAMP)) {
+            result.setIntegerStatistics(new IntegerStatistics(getTimestampValue(timeZone, min), getTimestampValue(timeZone, max)));
+        }
+        else if (type instanceof DecimalType) {
+            result.setDecimalStatistics(new DecimalStatistics(getDecimalValue(session, type, min), getDecimalValue(session, type, max)));
+        }
+        else {
+            throw new IllegalArgumentException("Unexpected type: " + type);
+        }
+    }
+
+    private static OptionalLong getIntegerValue(ConnectorSession session, Type type, Block block)
+    {
+        return block.isNull(0) ? OptionalLong.empty() : OptionalLong.of(((Number) type.getObjectValue(session, block, 0)).longValue());
```

Comment:
I think I know why `((Number) type.getObjectValue(session, block, 0)).longValue()` instead of `type.getLong(block, 0)` (because the second would work also for eg short decimal, skipping internal-representation-to-external conversion). While this method is called only for integral types, it's better as-is.
Consider leaving a comment hinting at the choice made